wrangler-service/src/main/java/io/cdap/wrangler/service/schema/SchemaViewerHandler.java

```diff
@@ -0,0 +1,219 @@
+      this.standard = standard;
+      this.version = version;
+    }
+  }
+
+  /**
+   * Populates the SchemaFileStore with the healthcare specific schema definitions.
+   * @param context the HTTP service runtime context
+   * @throws Exception
+   */
+  @Override
+  public void initialize(SystemHttpServiceContext context) throws Exception {
+    super.initialize(context);
+
+    try {
+      TransactionRunners.run(getContext(), ctx -> {
+        SchemaFileStore store = SchemaFileStore.get(ctx);
+        if (store.empty()) {
+          for (SchemaFileDescriptor descriptor : SCHEMA_LOADER.load()) {
+            store.updateSchema(descriptor);
+          }
+        }
+      });
+    } catch (Exception e) {
+      LOG.warn("Exception while loading default healthcare schemas ", e);
+    }
+  }
+
+  /**
+   * Returns a listing of the meta data for all available standards that have schema files.
+   * By default excludes the JsonSchema definitions. If the JsonSchema is required, the standard
+   * specific endpoint (../schemas/{id}) should be used.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchemas(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = SchemaFileStore.get(context);
+      List<SchemaListingResponse> listing = new ArrayList<>();
+      for (Map.Entry<String, SchemaFile.MetaData> entry :
+          store.listSchemaFileMetadata().entrySet()) {
+        SchemaFile.MetaData metaData = entry.getValue();
+        listing.add(new SchemaListingResponse(entry.getKey(),
+            metaData.getStandard(),
+            metaData.getVersion()));
+      }
+      return new ServiceResponse<>(listing);
+    }));
+  }
+
+  /**
+   * Returns the JsonSchema for the standard.
+   *
+   * @param request Handler for incoming request.
+   * @param responder Responder for data going out.
+   * @param namespace Namespace of request.
+   * @param id The id of the standard returned by the .../schemas listing.
+   */
+  @GET
+  @Path("contexts/{context}/schemaviewer/schemas/{id}")
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  public void getSchema(HttpServiceRequest request, HttpServiceResponder responder,
+      @PathParam("context") String namespace, @PathParam("id") String id) {
+    respond(request, responder, namespace, ns -> TransactionRunners.run(getContext(), context -> {
+      SchemaFileStore store = getStore(context, id);
+      return new ServiceResponse<>(store.readSchemaFileDescriptor(id));
+    }));
+  }
+
+  /**
+   * Returns all of the top level entities associated with a specific standard.
```

Comment:
a 'standard' is a JSON schema specific concept. Conceptually, this is just the list of top level field names of a schema.

Reply:
Changed terminology. Now using DataModels and models to better reflect intent.